src/expr/src/scalar/func.rs

```diff
@@ -1382,7 +1412,12 @@ fn div_float32<'a>(a: Datum<'a>, b: Datum<'a>) -> Result<Datum<'a>, EvalError> {
     if b == 0.0 && !a.is_nan() {
         Err(EvalError::DivisionByZero)
     } else {
-        Ok(Datum::from(a / b))
+        let quotient = a / b;
+        if quotient == 0.0 {
```

Comment:
Is this right? What about `0/1`?

Reply:
ha yes you are very correct.  I'll update this to match the behavior nikhil linked